### PR TITLE
MBS-11032: Convert Add medium edit to React 

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-11032

Converts MediumTracklist too. The TT for that contains a lot of code for relationship display, meant for release index pages as per the commit that added it (d2bdd1a). Release index pages don't use this code at all anymore though; only medium edits and cdtoc pages do (and neither of those display relationships at all). So I dropped all that code on the React version.

I moved the macros that are only used in add_medium.tt to just be part of AddMedium.js.

### CODE REVIEW

File: root/edit/details/AddMedium.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  startNumber,
+}: CondensedTrackACsRowProps): React.Element<'tr'> => (
+  <tr className={loopParity(rowCounter)}>
+    <td className="pos t">
+      {endNumber && endNumber !== startNumber
+        ? startNumber + '-' + endNumber
+        : startNumber}
+    </td>
+    <td>
+      <ExpandedArtistCredit artistCredit={artistCredit} />
```
Comment (yvanzo) [inline root/edit/details/AddMedium.js:62]: Note: Another change is that `ExpandedArtistCredit` does not insert extra `br` tags when there is nothing to expand, thus not all lines have the same height in the “Artist Credits” table.

File: root/edit/details/AddMedium.js [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  tracks,
+}: CondensedTrackACsProps):
+  Array<React.Element<typeof CondensedTrackACsRow>> => {
+  if (!tracks) {
+    return [];
+  }
+
+  let thisCredit;
+  let thisPosition = 0;
+  let rowCounter = 1;
```
Comment (yvanzo) [inline root/edit/details/AddMedium.js:77]: Odd/even row classes have been shifted by one row, this fixes it:

```suggestion
  let rowCounter = 0;
```

File: root/edit/details/AddMedium.js (hunk 1 above)
Comment (reosarevok) [reply root/edit/details/AddMedium.js:62]: That's not a bad thing though, is it?

Comment (yvanzo) [reply root/edit/details/AddMedium.js:62]: I have no idea whether initial height/new height are intentional or not, just noticing an undocumented change.

Comment (reosarevok) [reply root/edit/details/AddMedium.js:62]: Oh, ok. Pretty sure that was a bug in the expanded_artist_credits implementation in TT :)

**Pull request merged**